src/plugins/python/pythonplugin.cpp

```diff
@@ -238,33 +273,53 @@ bool PythonPlugin::loadOrReloadModule(ScriptEntry &script)
     if (!script.module)
         return false;
 
-    PyObject *pluginClass = findPluginSubclass(script.module);
-
-    if (!pluginClass) {
-        PySys_WriteStderr("Extension of tiled.Plugin not defined in "
-                          "script: %s\n", name.constData());
-        return false;
-    }
+    PyObject *pluginClass = findPluginSubclass(script.module, mTilesetPluginClass);
 
-    if (script.mapFormat) {
-        script.mapFormat->setPythonClass(pluginClass);
+    if(pluginClass) {
+      if (script.tilesetFormat) {
+          script.tilesetFormat->setPythonClass(pluginClass);
+      } else {
+          PySys_WriteStdout("---- Tileset plugin\n");
+          script.tilesetFormat = new PythonTilesetFormat(name, pluginClass, this);
+          addObject(script.tilesetFormat);
+      }
     } else {
-        script.mapFormat = new PythonMapFormat(name, pluginClass, this);
-        addObject(script.mapFormat);
+      PyObject *module = PyImport_ImportModule(name.constData());
+      Py_DECREF(script.module);
+      script.module = module;
+      pluginClass = findPluginSubclass(script.module, mPluginClass);
+      if (!pluginClass) {
+          PySys_WriteStderr("Extension of tiled.Plugin not defined in "
```

Comment:
Maybe adjust this to:
```suggestion
          PySys_WriteStderr("No extension of tiled.Plugin or tiled.TilesetPlugin defined in "
```